project_timesheet_management/models/timesheet_invoice.py

```diff
@@ -0,0 +1,764 @@
+                mail_template.sudo().with_context(state=mail_status).send_mail(
+                    timesheet_inv.id, force_send=True,
+                    email_values={"author_id": author_id or False})
+                u_lst = [timesheet_inv.pre_approved_by.id]
+            timesheet_inv.with_context(from_approval=True).sudo().write(
+                {"state": "rejected"}
+            )
+
+            # Project Manager
+            if (
+                timesheet_inv.project_id.user_id.id not in u_lst
+                and timesheet_inv.project_id.product_owner_id.id == self.env.user.id
+            ):
+                mail_template.write({
+                    'email_from': self.env.user.email or self.env.user.partner_id.email,
+                    'email_to': timesheet_inv.project_id.user_id.email or '',
+                    'reply_to': self.env.user.email or self.env.user.partner_id.email})
+                mail_template.sudo().with_context(state=mail_status).send_mail(
+                    timesheet_inv.id, force_send=True,
+                    email_values={"author_id": author_id or False})
+
+            # Supplier Project Coordinator
+            user_ids = timesheet_inv.project_id.favorite_user_ids.filtered(
+                lambda user: user.has_group(po_group) and
+                user.id not in u_lst and (
+                    user.partner_id == timesheet_inv.partner_id or
+                    user.supplier_company_id == timesheet_inv.partner_id)) or \
+                timesheet_inv.project_id.resource_ids.mapped(
+                    'user_ids').filtered(
+                        lambda user: user.has_group(po_group) and
+                user.id not in u_lst and (
+                    user.partner_id == timesheet_inv.partner_id or
+                    user.supplier_company_id == timesheet_inv.partner_id))
+    
+            for user_id in user_ids:
+                mail_template.write({
+                    'email_to': user_id.email or '',
+                    'email_from': self.env.user.email or self.env.user.partner_id.email,
+                    'reply_to': self.env.user.email or self.env.user.partner_id.email,})
+                mail_template.sudo().with_context(state=mail_status).send_mail(
+                    timesheet_inv.id, force_send=True,
+                    email_values={'author_id': author_id or False})
+
+            # Canna Project Coordinator
+            u_lst += user_ids.ids
+            c_user_ids = timesheet_inv.project_id.favorite_user_ids.filtered(
+                lambda user: user.has_group(
+                    cann_po_group) and user.id not in u_lst) or \
+                timesheet_inv.project_id.resource_ids.mapped(
+                    'user_ids').filtered(
+                    lambda user: user.has_group(
+                        cann_po_group) and user.id not in u_lst)
+            for user_id in c_user_ids:
+                mail_template.write({
```

Comment:
You should not write to the email template, instead you can use email_values as parameter for send_mail function and no need to define reply_to  separately as it uses the same value as email_from